### PR TITLE
Move esm samples to jsapi-resources

### PR DESCRIPTION
**Pull request by andygup**
Pending the 4.18 production release, this PR:
* Updates README
* Updates ESM issue template
* Removes ESM overview page
* Removes ESM samples (in preparation to move them over to jsapi-resources repo)

### CODE REVIEW

**Comment by andygup** [comment]
FYI @odoe (couldn't find you under Reviewers)

File: .github/ISSUE_TEMPLATE/esm.md [hunk 1]
```diff
@@ -32,14 +32,14 @@ A clear and concise description of what you expected to happen.
...
- - Browser: [e.g. "Edge 78 on Windows 10"]
+ - Browser: [e.g. "Chrome 87 on Windows 10"]
  
 **What's your OS? (check https://whatsmyos.com):**
  - OS: [e.g. "Apple iOS 12.4" or "Windows 10* 64-bit"]
  - Device: [e.g. "Windows desktop PC" or "iPhone 11"]
 
 **API version (check your console)**
- - Version: [e.g. ArcGIS API for JavaScript 4.13-next.YYYYMMDD [Revision: d8d62578]]
+ - Version: [Run `npm ls @arcgis/core` in a terminal window]
```
Comment (dasa) [inline .github/ISSUE_TEMPLATE/esm.md:42]: Should this be updated for both the CDN or either NPM package?

**Review by dasa — CHANGES_REQUESTED** [review]

**Review by dasa — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/esm.md (hunk 1 above)
Comment (dasa) [reply .github/ISSUE_TEMPLATE/esm.md:42]: Sorry, I see now that this file is specific for the `@arcgis/core` package.

File: .github/ISSUE_TEMPLATE/esm.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ STEP 1: Are you in the right place?
 A clear and concise description of what the bug is, and the exact error message.
 
 **Framework and build tools** 
- - Angular, React, node.js, vue, webpack, rollup
+ - Angular, React, node.js, vue, webpack, rollup [e.g. Angular 10.0.2]
```
Comment (odoe) [inline .github/ISSUE_TEMPLATE/esm.md:23]: Missing `,` or does this mean Angular `10.0.2` uses rollup?

**Review by odoe — COMMENTED** [review]
I just had the one question for clarification, rest looks good.

**Comment by bsvensson** [comment]
@andygup  We have a fair amount of traffic going to the ESM page, should we maybe leave a stripped down ESM overview page - with some info about (a) it has been released and (b) where to go to for the samples?

**Comment by andygup** [comment]
@bsvensson I wondered about that, as well. I'll add it back with some verbiage.

**Comment by andygup** [comment]
@bsvensson fyi, updated ES modules overview page text: https://github.com/Esri/feedback-js-api-next/pull/89/commits/962cb8ad6e55560105c725d2a67214f98129d991

**Pull request merged**